File: problem_2/problem_2.c
//Problem 2
//http://projecteuler.net/problem=2
//By considering the terms in the Fibonacci 
//  sequence whose values do not exceed four million, 
//  find the sum of the even-valued terms.

#include <stdio.h>

int main(int argc, char *argv[]){
  
  int prev = 1;
  int next = 2;
  int hold;

  int sum = 1;
  
  while(next < 4000000){
    
    hold = prev;
    prev = next;
    next = next + hold;
    
    if(next % 2 == 0){
      sum += next;
    }
   
  }
  printf("sum is %d",sum);  // 4613731

  return 0;
}

File: ruby/problem-2.rb
#Project Euler problem 2 
#By considering the terms in the Fibonacci sequence whose values do
#not exceed four million, find the sum of the even-valued terms.

fibs = Enumerator.new do |y|
  a = b = 1
  loop do
    y << a
    a, b = b , a + b
  end
end

puts fibs.take_while{ |n| n < 4000000 }.
  find_all{ |n| n.even? }.
  inject(0){ |sum , n| sum + n}
